packages/xrpl/src/models/methods/feature.ts

```diff
@@ -0,0 +1,68 @@
+import { BaseRequest, BaseResponse } from './baseMethod'
+
+export interface FeatureAllRequest extends BaseRequest {
+  command: 'feature'
+
+  feature?: never
```

Comment:
```suggestion
```

Can we remove this line?
I've seen `never` used in the context of infeasible control flows. Why do we need to state the absence of a field in an interface?

Reply:
It's necessary to distinguish the types. One weirdness about TS types is that you can add on extra fields and it'll still be that type. So this is still a valid `FeatureAllRequest` object according to TS without this field:
```typescript
{
  command: 'feature'
  feature: 'something'
}
```
Adding the `never` is needed to ensure that this isn't actually a valid `FeatureAllRequest` type.

The `?` is needed because for some reason you get errors for `'feature' field not included` in the `FeatureAllRequest` type if you don't have it.

Reply:
cc @justinr1234, our local expert on TS types

Reply:
Will look

Reply:
An example in our codebase: https://github.com/XRPLF/xrpl.js/blob/7e733c4446b9f2bc80b144bff03d46bdd5c9df93/packages/xrpl/src/models/ledger/BaseLedgerEntry.ts#L18-L26

Reply:
yes, I did come across other examples in the codebase

Reply:
I only saw control-flow related examples in the Typescript docs website: https://www.typescriptlang.org/docs/handbook/2/functions.html#never

It just feels strange that we are listing non-existent variables in the interface. But I get the rationale as a work-around to typescript's type inference.

This is not a strong objection, but it's nice to not have to do this

Reply:
I agree that it's kind of ugly, but I think it's necessary in TypeScript's typing system.